File: samples-java/src/main/java/io/github/ilaborie/loops/samples/anatomy/Transform.java
package io.github.ilaborie.loops.samples.anatomy;

import io.github.ilaborie.loops.samples.Element;

import java.util.ArrayList;
import java.util.List;

import static io.github.ilaborie.loops.samples.Elements.getSomeElements;

public class Transform {

    public static void main(String[] args) {
        List<Element> elements = getSomeElements();
        List<String> result = transform(elements);
        result.forEach(System.out::println);
    }

    public static List<String> transform(List<Element> elements) {
        List<String> result = new ArrayList<>();

        for (Element elt : elements) {
            String transformed = elt.getName() + " \\o/";
            result.add(transformed);
        }

        return result;
    }

}

File: samples-kotlin/src/main/kotlin/io/github/ilaborie/loops/samples/anatomy/transform.kt
package io.github.ilaborie.loops.samples.anatomy

import io.github.ilaborie.loops.samples.ElementKt
import io.github.ilaborie.loops.samples.getSomeElementsKt


fun main() {
    val elements = getSomeElementsKt()
    val result = io.github.ilaborie.loops.samples.stream.transform(elements)
    result.forEach { println(it) }
}

fun transform(elements: List<ElementKt>): List<String> {
    val result = mutableListOf<String>()

    for (elt in elements) {
        val transformed = elt.name + " \\o/"
        result.add(transformed)
    }

    return result
}